nes-network/network/src/sender/control.rs

```diff
@@ -0,0 +1,453 @@
+/*
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+use super::channel::{ChannelControlQueue, ChannelControlQueueListener, create_channel_handler};
+use crate::channel::{Channel, Communication};
+use crate::protocol::*;
+use crate::util::{ActiveTokens, ScopedTask};
+use futures::SinkExt;
+use std::collections::HashMap;
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio::sync::oneshot;
+use tokio_retry2::RetryError::Transient;
+use tokio_retry2::strategy::{ExponentialBackoff, jitter};
+use tokio_retry2::{Retry, RetryError};
+use tokio_stream::StreamExt;
+use tokio_util::sync::CancellationToken;
+use tracing::{Instrument, Span, error, info, info_span, warn};
+
+pub type Result<T> = std::result::Result<T, Error>;
+pub type Error = Box<dyn std::error::Error + Send + Sync>;
+pub(super) enum NetworkingServiceControlMessage {
+    RegisterChannel(
+        ConnectionIdentifier,
+        ChannelIdentifier,
+        oneshot::Sender<ChannelControlQueue>,
+    ),
+}
+pub(super) enum NetworkingConnectionControlMessage {
+    RegisterChannel(ChannelIdentifier, oneshot::Sender<ChannelControlQueue>),
```

Comment:
Please add a comment what the difference is between a `NetworkingServiceControlMessage::RegisterChannel` and a `NetworkingConnectionControlMessage::RegisterChannel`.

Reply:
added a comment